delete-publication/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+import static org.apache.http.HttpStatus.SC_INTERNAL_SERVER_ERROR;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+public class DeletePublicationHandlerTest {
+
+    public static final String IDENTIFIER = "identifier";
+    public static final String WILDCARD = "*";
+    private DeletePublicationHandler handler;
+    private PublicationService publicationService;
+    private Environment environment;
+    private OutputStream outputStream;
+    private Context context;
+    private static final ObjectMapper objectMapper = JsonUtils.objectMapper;
+
+    public static final JavaType PARAMETERIZED_GATEWAY_RESPONSE_VOID_RESPONSE_TYPE = objectMapper
+            .getTypeFactory()
+            .constructParametricType(GatewayResponse.class, Void.class);
+    public static final JavaType PARAMETERIZED_GATEWAY_RESPONSE_PROBLEM_TYPE = objectMapper
+            .getTypeFactory()
+            .constructParametricType(GatewayResponse.class, Problem.class);
+
+    @BeforeEach
+    public void setUp() {
+        publicationService = Mockito.mock(PublicationService.class);
+        environment = Mockito.mock(Environment.class);
+        when(environment.readEnv(ALLOWED_ORIGIN_ENV)).thenReturn(WILDCARD);
+        handler = new DeletePublicationHandler(publicationService, environment);
+        outputStream = new ByteArrayOutputStream();
+        context = Mockito.mock(Context.class);
+    }
+
+    @Test
+    void handleRequestReturnsAcceptedWhenServiceIsOk() throws IOException {
+        UUID identifier = UUID.randomUUID();
+        InputStream inputStream = new HandlerRequestBuilder<Publication>(objectMapper)
+                .withHeaders(TestHeaders.getRequestHeaders())
+                .withPathParameters(singletonMap(IDENTIFIER, identifier.toString()))
+                .build();
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<Void> gatewayResponse = toGatewayResponse();
+        assertEquals(HttpStatus.SC_ACCEPTED, gatewayResponse.getStatusCode());
+    }
+
+    @Test
+    void handleRequestReturnsBadRequestOnServiceInError() throws IOException, ApiGatewayException {
+        UUID identifier = UUID.randomUUID();
+        InputStream inputStream = new HandlerRequestBuilder<Publication>(objectMapper)
+                .withHeaders(TestHeaders.getRequestHeaders())
+                .withPathParameters(singletonMap(IDENTIFIER, identifier.toString()))
+                .build();
+        doThrow(InputException.class).when(publicationService).markPublicationForDeletion(identifier);
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        GatewayResponse<Problem> gatewayResponse = toGatewayResponseProblem();
+        assertEquals(HttpStatus.SC_BAD_REQUEST, gatewayResponse.getStatusCode());
+    }
+
+    private GatewayResponse<Void> toGatewayResponse() throws JsonProcessingException {
```

Comment:
Use the GatewayResponse.fromOutputStream() method. You need to declare the outputStream to be a ByteArrayOutputStream. It is OK to do so since we are using the ByteArrayOutputStream::toString method and not the generic OutputStream::toString method.